app/Http/Controllers/Three/ReactionController.php

```diff
@@ -0,0 +1,70 @@
+<?php
+
+namespace Rogue\Http\Controllers\Three;
+
+use Rogue\Models\Post;
+use Rogue\Models\Reaction;
+use Illuminate\Http\Request;
+use Rogue\Http\Controllers\Api\ApiController;
+use Rogue\Http\Transformers\Three\ReactionTransformer;
+
+class ReactionController extends ApiController
+{
+    /**
+     * @var \League\Fractal\TransformerAbstract;
+     */
+    protected $transformer;
+
+    /**
+     * Create a controller instance.
+     *
+     * @return void
+     */
+    public function __construct()
+    {
+        $this->transformer = new ReactionTransformer;
+
+        $this->middleware('auth.api');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param \Illuminate\Http\Request $request
+     * @param \Rogue\Models\Post $post
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request, Post $post)
+    {
+        $this->validate($request, [
+            'northstar_id' => 'required|string',
+        ]);
+
+        $userId = $request['northstar_id'];
+        // Check to see if the post has a reaction from this particular user with id of northstar_id. If not, create one.
```

Comment:
ah I see! I updated it and tested and seems to work! Is this what you meant @DFurnes ?

Reply:
Yeah, this seems great!